addon/components/frost-autocomplete-dropdown.js

```diff
@@ -0,0 +1,490 @@
+    }
+  },
+
+  /**
+   * Get necessary property values for positioning dropdown below select
+   * @param {Number} height - height of select
+   * @param {Number} top - top position of select
+   * @returns {Object} property values
+   */
+  _positionBelowInput (height, top) {
+    // Make sure dropdown is rendered below input and we leave space for arrow
+    // that connects dropdown to input
+    top = top + height - $(document).scrollTop() + 2
+
+    if (top === this.get('top')) {
+      return {}
+    }
+
+    return {
+      bottom: 'auto',
+      maxHeight: $(window).height() - top - WINDOW_SPACE,
+      top
+    }
+  },
+
+  /**
+   * Unbind event listeners to items in dropdown
+   * @param {HTMLElement} dropdownListElement - dropdown list element (ul)
+   */
+  _removeListItemEventListeners (dropdownListElement) {
+    const listItemElements = dropdownListElement.querySelectorAll('li')
+
+    Array.from(listItemElements).forEach((li, index) => {
+      $(li)
+        .off('mousedown')
+        .off('mouseenter')
+    })
+  },
+
+  /* eslint-disable complexity */
+  _updatePosition ($element) {
+    if (this.isDestroyed || this.isDestroying) return {}
+
+    $element = $element.first()
+
+    const {center, height, left, top, width} = this._getElementDimensionsAndPosition($element)
+    const windowCenterX = $(window).height() / 2 + $(document).scrollTop()
+    const props = (
+      center.x > windowCenterX ? this._positionAboveInput(top) : this._positionBelowInput(height, top)
+    )
+
+    if (left !== this.get('left')) {
+      props.left = left
+    }
+
+    if (width !== this.get('width')) {
+      props.width = width
+    }
+
+    return props
+  },
+  /* eslint-enable complexity */
+
+  _updateText () {
+    const filter = this.get('filter')
+    const dropdownListElement = document.getElementById('frost-select-list')
+    const clonedDropdownListElement = dropdownListElement.cloneNode(true)
+    const clonedTextElements = clonedDropdownListElement.querySelectorAll('.frost-select-list-item-text')
+    const textElements = dropdownListElement.querySelectorAll('.frost-select-list-item-text')
+    const scrollTop = dropdownListElement.scrollTop
+    const wrapLabels = this.get('wrapLabels')
+    const updateText = function (texElements, clonedTextElements) {
+      Array.from(texElements).forEach((textElement, index) => {
+        if (!wrapLabels) {
+          const clonedTextElement = clonedTextElements[index]
+          const updatedData = trimLongDataInElement(clonedTextElement)
+
+          if (updatedData) {
+            textElement.textContent = updatedData.text
+            textElement.setAttribute('title', updatedData.tooltip)
+          }
+        }
+
+        if (filter) {
+          const pattern = new RegExp(filter.replace(/[.*+?^${}()|[\]\\]/, '\\$&'), 'i')
+          const textWithMatch = textElement.textContent.replace(pattern,
+            "<span class='frost-select-list-item-text-highlight'>$&</span>")
```

Comment:
Is hard-coded HTML and `innerHTML` the best approach here?  We can't add a new node to an element and apply/remove classes?  This approach has a "smell" to it.